mpa/modules/models/detectors/sam_detector_mixin.py

```diff
@@ -19,22 +19,33 @@ def train_step(self, data, optimizer, **kwargs):
         self.current_batch = data
         return super().train_step(data, optimizer, **kwargs)
 
-    def simple_test(self, img, img_metas, rescale=False, postprocess=True):
-        x = self.extract_feat(img)
-        outs = self.bbox_head(x)
-        with no_nncf_trace():
-            bbox_results = \
-                self.bbox_head.get_bboxes(*outs, img_metas, self.test_cfg, False)
-            if torch.onnx.is_in_onnx_export():
-                feature_vector = get_feature_vector(x)
-                cls_scores = outs[0]
-                saliency_map = DetSaliencyMapHook(self).func(cls_scores, cls_scores_provided=True)
-                feature = feature_vector, saliency_map
-                return bbox_results[0], feature
+    def simple_test(self,
+                    img,
+                    img_metas,
+                    proposals=None,
+                    rescale=False,
+                    postprocess=True):
+        """
+        Class-wise Saliency map for Single-Stage Detector, otherwise use class-ignore saliency map.
+        """
+        if isinstance(self, TwoStageDetector):
+            return super().simple_test(img, img_metas, proposals, rescale, postprocess)
+        else:
+            x = self.extract_feat(img)
+            outs = self.bbox_head(x)
+            with no_nncf_trace():
+                bbox_results = \
+                    self.bbox_head.get_bboxes(*outs, img_metas, self.test_cfg, False)
+                if torch.onnx.is_in_onnx_export():
+                    feature_vector = get_feature_vector(x)
+                    cls_scores = outs[0]
+                    saliency_map = DetSaliencyMapHook(self).func(cls_scores, cls_scores_provided=True)
+                    feature = feature_vector, saliency_map
+                    return bbox_results[0], feature
 
-        if postprocess:
-            bbox_results = [
-                self.postprocess(det_bboxes, det_labels, None, img_metas, rescale=rescale)
-                for det_bboxes, det_labels in bbox_results
-            ]
-        return bbox_results
+            if postprocess:
+                bbox_results = [
+                    self.postprocess(det_bboxes, det_labels, None, img_metas, rescale=rescale)
+                    for det_bboxes, det_labels in bbox_results
+                ]
+            return bbox_results
```

Comment:
I might have missed the code review for this part.
Could you explain why XAI feature is implemented in "SAM" mixin rather than implementing "XAIMixin"?
It's a refactoring issue, not mandatory fix in this PR, though :)

Reply:
I think this would be because the saliency map exporting is subordinate to the inference step. How's your thought @negvet ?

Reply:
I agree actually that it make sense to implement additional `Mixin` for XAI purposes (if it is ok for us to have one more inheritance, is it?).

But for the detection I followed the classification convention:
https://github.com/openvinotoolkit/model_preparation_algorithm/blob/a99aea0296a6c1987c014d55c26dfb56ca8be611/mpa/modules/models/classifiers/sam_classifier.py#L274
which have been like this even before default XAI algo replacement.